Hakawai/Mentions/HKWMentionsPlugin.m

```diff
@@ -848,6 +848,29 @@ - (void)toggleAutocorrectAsRequiredForRange:(NSRange)range {
 // JIRA: POST-13757
 #pragma mark - Mention Query Utils
 
+/**
+ Find the location for the end of the next word, starting at the given location. If there is an invalid character (such as a mention), return @c NSNotFound
+
+ @param location location to begin search from
+ @param text text to search
+ @return location of end of next word, start at @c location
+ */
+- (NSUInteger)endOfValidWordAfterLocation:(NSUInteger)location text:(nonnull NSString *)text {
```

Comment:
- can this be a class method too? 
- worth clarifying what word is considered as `valid` as it's not obvious. is it just non-Mention?
- can we always put text as first param just like other methods? 
- move this below `- (NSString *)mentionsQuery:(NSString *)text location:(NSUInteger)location` also should the function below be named something like: `- (NSString *)mentionsQueryWithinText:(NSString *)text location:(NSUInteger)location`

Reply:
- probably out of scope for this PR, let's create a utils file to have those methods instead to keep mentionsPlugin lightweight when you finish creating V2 version

Reply:
1. it's non-trivial because of the nature of the self method used. Will wrap it into the general util method work.
2. "If there is an invalid character (such as a mention)" is already there. Changed the comment to make it clearer
3. done
4. done and done